backend/src/main/scala/bloop/ScalaInstance.scala

```diff
@@ -60,7 +68,25 @@ final class ScalaInstance private (
     filename.startsWith(ScalacCompilerName) ||
       (isDotty && (filename.startsWith("dotty-compiler") || filename.startsWith("scala3-compiler")))
   private def hasScalaLibraryName(filename: String): Boolean =
-    filename.startsWith("scala-library")
+    filename.startsWith("scala-library") || filename.startsWith("scala3-library")
```

Comment:
I really do not like how it works... Maybe we can change bloop options at some point to pass compiler jars more directly?

Reply:
It's not perfect, but we would need to split the Scala configuration to have two lists and we would most likely move this part to the integrations (which might also just have a single Scala classpath). Overall I feel it's better to keep it here and besides `allJars` are only Scala jars, so no user defined libraries will pop up here.

The only risk is if someone forks the compiler, but then a lot more things could break.